mobile/android/android-emulator-services.ts

```diff
@@ -410,10 +410,17 @@ class AndroidEmulatorServices implements Mobile.IAndroidEmulatorServices {
 		return result;
 	}
 
-	private getBestFit(): string {
+	private getBestFit(suggestedImage?: string): string {
 		let minVersion = this.$emulatorSettingsService.minVersion;
 
-		let best = _(this.getAvds())
+		let avdResults = this.getAvds();
+		if (suggestedImage && !_.includes(avdResults, suggestedImage)) {
```

Comment:
The whole if/else can be simplified even further to just one filter:

if(suggestedImage) {
   avgResults = avgResults.filter(avd => avd === suggestedImage);
}